src/main/java/com/okta/sdk/framework/RateLimitContext.java

```diff
@@ -0,0 +1,58 @@
+package com.okta.sdk.framework;
+
+import com.okta.sdk.exceptions.SdkException;
+import org.apache.http.Header;
+import org.apache.http.HttpResponse;
+import org.joda.time.DateTime;
+
+public class RateLimitContext {
+    private HttpResponse httpResponse;
+
+    public RateLimitContext(HttpResponse httpResponse) {
+        this.httpResponse = httpResponse;
+    }
+
+    private String getHeaderValueString(String headerName) throws Exception {
+        if (httpResponse == null) {
+            throw new SdkException("No http response");
+        }
+
+        Header[] headers = httpResponse.getHeaders(headerName);
+        if (headers.length > 0) {
+            Header header = headers[0];
+            return header.getValue();
+        } else {
+            throw new SdkException("No " + headerName + " header");
+        }
+    }
+
+    private Long getHeaderValueLong(String headerName) throws Exception {
+        String headerString = getHeaderValueString(headerName);
+        try {
+            return Long.parseLong(headerString);
+        } catch (Exception e){
+            throw new SdkException("Error parsing " + headerName + " header");
+        }
+    }
+
+    public Long getNumRequestsRemaining() throws Exception {
```

Comment:
Shouldn't we have detailed Javadoc on each of these explaining what they mean? It's not obvious just from reading the method names how exactly rate limiting works.